tests/bluechi_test/test.py

```diff
@@ -77,11 +85,20 @@ def setup(self) -> Tuple[bool, Tuple[BluechiControllerContainer, Dict[str, Bluec
                 node = BluechiNodeContainer(c, cfg)
                 node_container[cfg.node_name] = node
 
+                if self.run_with_valgrind:
+                    node.enable_valgrind()
+
                 node.exec_run('systemctl start bluechi-agent')
+                node.wait_for_unit_state_to_be("bluechi-agent.service", "active")
 
         except Exception as ex:
             success = False
             logger.debug(f"Failed to setup bluechi container: {ex}")
+            traceback.print_exc()
+
+        if self.run_with_valgrind:
+            # Give some more time for bluechi to start and connect while running with valgrind
+            time.sleep(2)
```

Comment:
I see there is a change in the default timer: in the function above from 2.0 to float = 4.0. Should we set 4 here too?

Reply:
No. The default timer that was changed from 2s to 4s is for timeout in `wait_for_unit_state_to_be()`. This is the max time that we are waiting for the service to become active, usually it takes even less time, but when running with valgrind I have seen that we reach this timeout sometimes, so I extended it to 4s. It does not mean that we wait 4 seconds.

The reason I have added additional 2 seconds _after_ bluechi services were started in all containers is that I have seen that when running with valgrind, even when the services are marked active - the agents are not necessarily already connected to the controller, which causes errors down the flow. Here we start bluechi services in all containers, and then add additional 2 seconds just to make sure everything is connected

Reply:
One could ask why I didn't use `systemctl_start_and_wait` here. Here we want to start all required bluechi services in all containers. No need to wait _between_ each `systemctl start` - we want to start everything, and only then take a little sleep to allow agents to connect.
Otherwise, in tests with a lot of agents we would add sleep time to each one of them, which would sum up to quite a lot of unnecessary sleep.